lib/elasticsearch.js

```diff
@@ -0,0 +1,540 @@
+
+		if(!Array.isArray(model.constructor.fields))
+			throw new ElasticSearchError('Invalid fields. it must be an array, see model.constructor.fields.', ElasticSearchError.codes.INVALID_MODEL);
+
+	}
+
+	/**
+	 * Build elasticsearch index with mappings for sortings
+	 * @param {Model} model Model instance
+	 */
+	async buildIndex(model) {
+
+		this.validateModel(model);
+
+		try {
+
+			const client = this.getClient();
+			const exists = await client.indices.exists({ index: model.constructor.table });
+
+			if(!exists.body)
+				await client.indices.create({ index: model.constructor.table });
+
+			await client.indices.putMapping({
+				index: model.constructor.table,
+				type: '_doc',
+				body: this.getMappingsFromModel(model)
+			});
+
+		} catch(err) {
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Analyses the model fields and build the mappings query for elasticsearch
+	 * @param {Model} model Model instance
+	 * @returns {Object} elasticsearch mappings query
+	 */
+	getMappingsFromModel(model) {
+
+		this.validateModel(model);
+
+		const parsedMappings = { properties: {} };
+
+		for(const field of model.constructor.fields) {
+			if(typeof field === 'object' && !Array.isArray(field)) {
+				Object.entries(field).forEach(([key, value]) => {
+					parsedMappings.properties[key] = {
+						type: value,
+						fields: {
+							raw: {
+								type: value
+							}
+						}
+					};
+				});
+				continue;
+			}
+
+			parsedMappings.properties[field] = {
+				type: 'text',
+				fields: {
+					raw: {
+						type: 'keyword'
+					}
+				}
+			};
+		}
+
+		if(!parsedMappings.properties.id) {
+			parsedMappings.properties.id = {
+				type: 'text',
+				fields: {
+					raw: {
+						type: 'keyword'
+					}
+				}
+			};
+		}
+
+		parsedMappings.properties.dateCreated = {
+			type: 'date',
+			fields: {
+				raw: {
+					type: 'date'
+				}
+			}
+		};
+
+		parsedMappings.properties.lastModified = {
+			type: 'date',
+			fields: {
+				raw: {
+					type: 'date'
+				}
+			}
+		};
+
+		return parsedMappings;
+	}
+
+	/**
+	 * Prepare the items for elasticsearch operations
+	 * @param {Array} items items
+	 */
+	prepareFields(items) {
+
+		if(!Array.isArray(items))
+			items = [items];
+
+		for(const item of items) {
+
+			if(!item.id)
+				item.id = UUID();
+
+		}
+	}
+
+	/**
+	 * Parse the filters into elasticsearch filters
+	 * @param {Object} filters filters
+	 * @returns {Object} elasticsearch filters
+	 */
+	parseFilters(filters) {
+		return ElasticSearchFilters.getFilters(filters);
+	}
+
+	/**
+	 * Parse the item into elasticsearch script string for updates
+	 * @param {Object} item item
+	 * @returns {Object} parsed item for updates
+	 */
+	parseValuesForUpdate(item) {
+
+		let parsedItem = '';
+
+		Object.entries(item).forEach(([key, value]) => {
+			parsedItem += `ctx._source.${key}='${value}';`;
```

Comment:
Para que no convierta el valor siempre por `string` el valor a actualizar
```suggestion
			parsedItem += `ctx._source.${key}=${value};`;
```